EIPS/eip-1620.md

```diff
@@ -1,296 +1,185 @@
+
+This standardised interface takes a stab at changing the way we think about trust in financial contracts. Thanks to blockchains, payments need not be sent in lumps (as with monthly salaries), as there is less overhead in pay-as-you-go. Money as a function of time would better align incentives in a host of scenarios.
 
 ### Use Cases
 
-This is just a preliminary list of use cases. There are other spooky ideas interesting to explore, such as time-dependent disincetivisation, but, for brevity, we have not included them here.
+This is just a preliminary list of use cases. There are other interesting ideas left to explore, such as time-dependent disincetivisation, but, for brevity, we have not included them here.
 
 - Salaries
+- Insurance
 - Subscriptions
 - Consultancies
-- CDPs
 - Rent
 - Parking
+- Lending
 
-### Crowdsales
-[RICOs](https://github.com/lukso-network/rico), or Reversible ICOs, were introduced at Devcon4 by @frozeman. The idea is to endow investors with more power and safety guarantees by allowing them to "reverse" the investment based on the evolution of the project. We previously discussed a similar concept called SICOs, or Streamable ICOs, in this research [thread](https://ethresear.ch/t/chronos-a-quirky-application-proposal-for-plasma/2928/14?u=paulrberg).
-
-Instead of investing a lump sum and giving the money away to the project developers, funds are held in a smart contract which allocates money based on the passage of time. Project developers can withdraw funds as the stream stays active, while investors have the power to get back a significant percentage of their initial commitment if the project halts.
+[Sablier](https://sablier.finance) is a champion project for this ERC. It is what we call the protocol for real-time finance.
 
 ## Specification
 
 ### Structs
 
-The structure of a `stream` should be as follows:
-
-- `stream`
-    - `sender`: the `address` of the entity funding the stream
-    - `recipient`: the `address` where the money is being delivered to
-    - `tokenAddress`: the `address` of the ERC20 token used as payment asset
-    - `balance`: the total funds left in the stream
-    - `timeframe`: as defined below
-    - `rate`: as defined below
-
-```solidity
-  struct Stream {
-    address sender;
+The structure of a stream should be as follows:
+
+- `Stream`
+    - `deposit`: the amount of money to be streamed
+    - `ratePerSecond`: the number of tokens allocated each second to the recipient
+    - `remainingBalance`: the left in the stream
+    - `startTime`: the unix timestamp for when the stream starts
+    - `stopTime`: the unix timestamp for when the stream stops
+    - `recipient`: the address towards which the money is streamed
+    - `sender`: the address of the party funding the stream
+    - `tokenAddress`: the ERC20 token to use as streaming currency
```

Comment:
ERC20 should be properly formatted and be a link, likewise with any other link to the EIP. This can be sorted in this PR or subsequent PRs while in last call.